task_processing/plugins/mesos/execution_framework.py

```diff
@@ -131,101 +220,18 @@ def _background_check(self):
-                        log.warning(
-                            f'Re-enqueuing task {task_id} in unknown state for '
-                            f'longer than {self.task_staging_timeout_s}'
-                        )
-                        # Re-enqueue task
-                        self.enqueue_task(md.task_config)
-                        get_metric(
-                            metrics.TASK_FAILED_TO_LAUNCH_COUNT).count(1)
-                        continue
-
-                    if md.task_state == 'TASK_STAGING':
-                        log.warning(f'Killing stuck task {task_id}')
-                        self.kill_task(task_id)
-                        self.task_metadata = self.task_metadata.set(
-                            task_id,
-                            md.set(
-                                task_state='TASK_STUCK',
-                                task_state_history=md.task_state_history.set(
-                                    'TASK_STUCK', time.time()),
-                            )
-                        )
-                        self.blacklist_slave(
-                            agent_id=self.task_metadata[task_id].agent_id,
-                            timeout=self.slave_blacklist_timeout_s,
-                        )
-                        get_metric(metrics.TASK_STUCK_COUNT).count(1)
-
-                    if md.task_state == 'TASK_STUCK':
-                        killed_at = md.task_state_history['TASK_STUCK']
-                        if time.time() - killed_at > 3600:
-                            log.warning(
-                                f'Task {task_id} waiting for terminal state '
-                                'for an hour, discarding from metadata'
-                            )
-                            self.task_metadata.discard(task_id)
-                            self.event_queue.put(
-                                task_event(
-                                    task_id=task_id,
-                                    terminal=True,
-                                    timestamp=time_now,
-                                    success=False,
-                                    message='stop',
-                                    task_config=md.task_config,
-                                    raw='Stuck and timed out',
-                                )
-                            )
-                        else:
-                            log.warning(f'Task {task_id} waiting for terminal state')
+                    self._background_check_task(
+                        time_now,
+                        tasks_to_reconcile,
+                        task_id,
+                        md,
+                    )
 
             self._reconcile_tasks(
                 [Dict({'task_id': Dict({'value': task_id})}) for
                     task_id in tasks_to_reconcile]
             )
+            log.info(f'background check done in {time.time() - time_now}s')
```

Comment:
+1 This is nice. Can you also add a metric for this? We can create a dashboard to track this.

Reply:
good point! done